new-explorer-frontend/src/utils/MainApi.js

```diff
@@ -0,0 +1,54 @@
+class MainApi {
+  constructor({ baseUrl, authorizationCode }) {
+    this._authorization = authorizationCode;
+    this._baseUrl = baseUrl;
+  }
+
+  _checkResponse(res) {
+    if (!res.ok) {
+      return Promise.reject(res.status);
+    }
+    return res.json();
+  }
+
+  getSavedArticles() {
+    return fetch(`${this._baseUrl}/articles`, {
+      headers: { Authorization: this._authorization },
+    }).then(this._checkResponse);
```

Comment:
**Might be better**

You can make a special method for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse)
  }

```

And now you can safely replace `fetch` with `this._request`. And you don’t need to change anything else.